src/view/actions/syncIdentity.jsx

```diff
@@ -16,34 +16,56 @@ import { array, object, string } from "yup";
 import "@react/react-spectrum/Form"; // needed for spectrum form styles
 import render from "../render";
 import ExtensionView from "../components/extensionView";
-import "./setCustomerIds.styl";
-import CustomerIdWrapper from "../components/customerIdWrapper";
-import getDefaultCustomerId from "../utils/getDefaultCustomerId";
+import IdentityWrapper from "../components/identityWrapper";
+import getDefaultIdentity from "../utils/getDefaultIdentity";
+import "./syncIdentity.styl";
+
+const identitiesMapToArray = identityMap => {
+  return Object.keys(identityMap).map(namespace => {
+    return {
+      namespace,
+      ...identityMap[namespace]
+    };
+  });
+};
+
+const identitiesArrayToMap = identitiesArray => {
+  return identitiesArray.reduce((identityMap, identity) => {
+    const { namespace, ...otherProperties } = identity;
+    identityMap[identity.namespace] = otherProperties;
+    return identityMap;
+  }, {});
+};
```

Comment:
This is the magic where we translate between our application state structure and our settings structure.